src/components/Snippet/SnippetTool/SnippetTool.js

```diff
@@ -1,19 +1,82 @@
+import { useState } from "react";
+import { useHistory } from "react-router";
 import styled from "styled-components";
 
 import Button from "../../common/Button";
+import SnippetSavingModal from "../../SnippetSavingModal";
+
+import { deleteSnippet } from "../../../api/service";
 
 import { TOOL } from "../../../constants/variants";
+import { OK } from "../../../constants/messages";
+
+import {
+  COPY,
+  SAVE,
+  SHARE,
+  DELETE
+} from "../../../constants/names";
 
 const ToolWrapper = styled.div`
   display: flex;
   justify-content: space-between;
   width: 320px;
 `;
 
-export default function SnippetTool() {
+export default function SnippetTool({ creator, language, code, snippetId }) {
+  const history = useHistory();
+  const [isOpened, setIsOpened] = useState(false);
+
+  const handleClick = async (event) => {
+    const buttonName = event.target.textContent;
```

Comment:
저는 구조 분해 할당으로 받는 스타일을 사용하고 있는데, 이 부분에 대해서는 팀원 모두와 상의한 후에 결정하면 좋을 것 같아요!

```suggestion
  const handleClick = async ({ target }) => {
    const buttonName = target.textContent;
```